### ISSUE TITLE
[rush] console.warn() output gets colored red

### ISSUE THREAD

**Issue by aeulitz** [microsoft/rushstack#1132]
It seems that stream-collator writes a child process' stderr output colored in red to its own process' stdout:

node_modules\@microsoft\stream-collator\lib\Interleaver.js(131)
    static _writeTask(taskName, taskInfo) {
        taskInfo.state = TaskWriterState.Written;
        if (!taskInfo.quietMode) {
            this._stdout.write(taskInfo.stdout.join(''));
        }
        this._stdout.write(colors.red(taskInfo.stderr.join(''))); // <<<<<<<<<<<<<<
    }
}

The call stack leading to the above method is shown below. As console.warn() is being written to stderr, it means that messages intended to be warnings visually appear like errors. This behavior seems to be intermittent and perhaps depending on the number of projects that are being built in parallel. In one instance of the problem, the warnings ended up in red only if the project in which the warnings originated was a build dependency of the "root project" specified via the '--to' option.

Callstack:
_writeTask (...\node_modules\@microsoft\stream-collator\lib\Interleaver.js:138)
_tasks.forEach (...\node_modules\@microsoft\stream-collator\lib\Interleaver.js:124)
_writeAllCompletedTasks (...\node_modules\@microsoft\stream-collator\lib\Interleaver.js:122)
_completeTask (...\node_modules\@microsoft\stream-collator\lib\Interleaver.js:112)
close (...\node_modules\@microsoft\stream-collator\lib\Interleaver.js:49)
taskPromises.push.task.execute.then (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:174)
_tickCallback (next_tick.js:188)
[ Promise.resolve ] (Unknown Source:undefined)
_startAvailableTasks (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:169)
taskPromises.push.task.execute.then.catch.then (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
_tickCallback (next_tick.js:188)
[ Promise.resolve ] (Unknown Source:undefined)
_startAvailableTasks (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
taskPromises.push.task.execute.then.catch.then (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
_tickCallback (next_tick.js:188)
[ Promise.resolve ] (Unknown Source:undefined)
_startAvailableTasks (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
taskPromises.push.task.execute.then.catch.then (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
_tickCallback (next_tick.js:188)
[ Promise.resolve ] (Unknown Source:undefined)
_startAvailableTasks (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
taskPromises.push.task.execute.then.catch.then (...\node_modules\@microsoft\rush-lib\lib\logic\taskRunner\TaskRunner.js:198)
_tickCallback (next_tick.js:188)

**Comment by octogonz** [comment]
This sounds like a bug.  I think we should fix it.

@nickpape-msft FYI

**Comment by iclanton** [comment]
We should talk about what the design should be here. Should it not modify the color at all?

**Comment by octogonz** [comment]
Agreed, tampering with colors seems unrelated to the functiality of correlating streams. It seems like it could potentially cause trouble. This is old code that wasn't analyzed very deeply. I'd suggest just to open a PR and fix it.

**Comment by elliot-nelson** [comment]
Fixed in a revamp of stream-collator last year.  If you experience this again, please open a new issue.
